XPCBackupService/Services/BackupUploadService.swift

```diff
@@ -150,36 +189,74 @@ struct BackupUploadService {
+                }
 
-            let encryptedFilename = try encrypt.encrypt(
-                string: filename.deletingPathExtension,
-                password: "\(config.CRYPTO_SECRET2)-\(stringRemoteParentId)",
-                salt: cryptoUtils.hexStringToBytes(config.MAGIC_SALT_HEX),
-                iv: Data(cryptoUtils.hexStringToBytes(config.MAGIC_IV_HEX))
-            )
+                let updatedFile = try await backupNewAPI.replaceFileId(
+                    fileUuid: remoteUuid,
+                    newFileId: result.id,
+                    newSize: result.size
+                )
+
+                self.logger.info("✅ Updated file correctly with identifier \(updatedFile.fileId)")
+
+                node.progress.completedUnitCount = 1
 
-            let createdFile = try await backupAPI.createBackupFile(
-                createFileData: CreateFileData(
-                    fileId: result.id,
-                    type: filename.pathExtension,
-                    bucket: result.bucket,
-                    size: result.size,
-                    folderId: remoteParentId,
-                    name: encryptedFilename.base64EncodedString(),
-                    plainName: filename.deletingPathExtension
+                // Edit date in synced database
+                try self.editSyncedNodeDate(remoteUuid: remoteUuid, date: Date.now)
+
+                if encryptedContentURL != nil {
+                    try FileManager.default.removeItem(at: encryptedContentURL!)
+                }
+
+                return (remoteId, remoteUuid)
+            } else {
+                // CREATE FILE REFERENCE
+                let stringRemoteParentId = "\(remoteParentId)"
+
+                let encryptedFilename = try encrypt.encrypt(
+                    string: filename.deletingPathExtension,
+                    password: "\(config.CRYPTO_SECRET2)-\(stringRemoteParentId)",
+                    salt: cryptoUtils.hexStringToBytes(config.MAGIC_SALT_HEX),
+                    iv: Data(cryptoUtils.hexStringToBytes(config.MAGIC_IV_HEX))
                 )
-            )
 
-            self.logger.info("✅ Created file correctly with identifier \(createdFile.id)")
+                let createdFile = try await backupAPI.createBackupFile(
+                    createFileData: CreateFileData(
+                        fileId: result.id,
+                        type: filename.pathExtension,
+                        bucket: result.bucket,
+                        size: result.size,
+                        folderId: remoteParentId,
+                        name: encryptedFilename.base64EncodedString(),
+                        plainName: filename.deletingPathExtension
+                    )
+                )
+                self.logger.info("✅ Created file correctly with identifier \(createdFile.id)")
+
+                node.progress.completedUnitCount = 1
+
+                // Save created file into synced database
+                try self.addSyncedNodeToDB(
```

Comment:
Same, move this to an independent service, BackupUploadService should only take care of uploading the file, if your method makes more than one thing, you are breaking the separation of concerns principle + this will be way more hard to test